### ISSUE TITLE
Allow seeing the RAML spec

### ISSUE THREAD

**Issue by ddossot** [mulesoft/api-console#122]
It would be great to have the option to display the RAML spec currently loaded by the console. If this view could be nicely formatted and syntax coloured, it would be even better :wink:

**Comment by usarid** [comment]
Yes, this was actually in the plan from day 1 of the original console and designer, in the form of a readonly designer that slides out of the side of the console as a "shelf". We've not yet implemented it, though we have prototyped something similar. But we have not forgotten and it will happen at some point... ;-)





On Tue, Mar 10, 2015 at 9:45 AM -0700, "David Dossot" <notifications@github.com> wrote:












It would be great to have the option to display the RAML spec currently loaded by the console. If this view could be nicely formatted and syntax coloured, it would be even better

**Comment by sichvoge** [comment]
+1 and additionally an option to propagate the RAML files to download it via the API Console

**Comment by MMore** [comment]
+1 including the previous comment.

**Comment by robvanmieghem** [comment]
+1

**Comment by tjsnell** [comment]
+1

**Comment by jarrodek** [comment]
I'm not convinced to the idea of having RAML source display / download option.

It might be OK for standalone API Console. But the console is also used in other products (like the designer) that may have some visual element for the RAML content. In this case it would be confusing for the user why this option is enabled.

As for Console v4 it doesn't even know anything about RAML because it accepts JavaScript object from parser and the enhancer. 
There's another problem of resolving the dependencies of RAML (fragments). It can be resolved from various sources (local or remote path). Is console should do this as well? That would be quite complex function in the console.

**Comment by usarid** [comment]
The Console needs to have an option to download the source RAML file(s), which are the same ones the parser is given. It can be disabled by an embedder if needed. The Console is, in terms of packaging, responsible for accepting a RAML file(s) and doing what's needed to create itself, including invoking the parser. In any case, allowing the end user to access the source spec is necessary, as that's the source of truth, and we want to make sure spec-oriented approaches are promoted.

**Comment by jarrodek** [comment]
I hear you and I understand the use case.
Option to disable viewer is OK. Though, I see few technical challenges to actually do it right. I have to plan this.

**Comment by sichvoge** [comment]
I'll have a backlog item for this.

**Comment by usarid** [comment]
When can we get a date for delivery?

**Comment by sichvoge** [comment]
We meet tomorrow to discuss the scope for the next releases and after that Pawel will look into the dates. Please bear with us.
